docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+Remember to mention the purpose of your request under "Any other comments/queries". 
+
+### Submitting a pull request
+
+We accept patches/fixes in form of pull requests (PRs). Make sure that the pull request addresses an open issue in the issue tracker.
+Remember to follow the [TEAMMATES Development Process](process.md).
+
+If this is your first time contributing to TEAMMATES, you may want to read up the contributor orientation guide lined out in the next section.
+
+## Contributor orientation guide
+
+Contributing to an OSS project requires you to figure out things on your own when you can, and seek help from the right resource (Google, StackOverflow, troubleshooting guides, issue tracker, etc.) when you cannot. To become a TEAMMATES contributor, you need to start honing those skills. To help you with that, we have created a sequence of tasks you can try to complete. Try to complete as many of them as you can, in the order they are listed.
+
+Of course we are happy to guide you if you encounter any difficulties when doing these tasks; we have provided a [troubleshooting guide](troubleshooting-guide.md) for some commonly faced problems. Failing that, you may post a help request in our [issue tracker](https://github.com/TEAMMATES/teammates/issues).
+
+> - The task descriptions are brief by intention. We want you to try to figure out how to do those things by yourself.
+> - While the dev team members will be glad to answer your questions, you will not make a good impression if you ask questions which are already answered in the mentioned resources.
+
+### Knowledge required
+
+Although TEAMMATES uses many tools and technologies, you need not know all of them before you can contribute. The diagram below shows which technologies you need to learn to contribute in different roles.
+
+![RolesAndTechnologies.png](images/RolesAndTechnologies.png)
+
+Note that these roles are not fixed or formally assigned; it is simply for guidance only.
+
+Roles:
+
+* **Web page developer**: Works on static web pages, such as those used in the TEAMMATES product web site.
+* **Front end developer**: Works on the front end of the app that is generated dynamically from the server side.
+* **Test developer**: Works on automating system tests.
+* **Back end developer**: Works on the back end logic of the app, including data storage.
+* **DevOps developer**: Works on automations (CI, static analysis, scripting) and operations (logging, monitoring, troubleshooting).
+
+### Orientation task list
+
+#### Phase A: Set up locally
+
+1. [Set up TEAMMATES development environment on your computer.](setting-up.md)<br>
+   **Important: Follow instructions to the letter. Install the specified versions of the tool stack, not the latest versions.**
+1. Get the tests to pass (more info in [_Development Guidelines_ document](development.md)).
+   It is OK to proceed to the next phase if you have fewer than 5 failing test cases.
+
+#### Phase B: Learn
```

Comment:
@whipermr5 I actually want to add something like a list of checkpoints here, something like

- [ ] Can add new instructors
- [ ] Can add new students
- [ ] Can do a simple UI tinker
- [ ] ...

But I can't really come up with a list that is not too long. Do you have suggestions? This needs not be now; it can be for the last PR that will finally close the issue.